src/aspire/source/micrograph.py

```diff
@@ -0,0 +1,306 @@
+        if particles_per_micrograph * micrograph_count > simulation.n:
+            raise ValueError(
+                "The particle count times micrograph count must be less than or equal to the number of images generated by the simulation (particles_per_micrograph * micrograph_count <= simulation.n)"
+            )
+        if simulation.L > micrograph_size:
+            raise ValueError(
+                "The micrograph size must be larger or equal to the simulation's image size."
+            )
+
+        self.particle_box_size = simulation.L
+        self.particles_per_micrograph = particles_per_micrograph
+        self.total_particle_count = (
+            self.micrograph_count * self.particles_per_micrograph
+        )
+        self.dtype = simulation.dtype
+
+        self.noise_adder = noise_adder
+
+        if boundary is None:
+            self.boundary = self.particle_box_size // 2
+        else:
+            if (
+                boundary < (0 - self.particle_box_size // 2)
+                or boundary > self.micrograph_size // 2
+            ):
+                raise RuntimeError("Illegal boundary value.")
+            self.boundary = boundary
+
+        if interparticle_distance is None:
+            self.interparticle_distance = self.particle_box_size
+        else:
+            self.interparticle_distance = interparticle_distance
+
+        # Create the radial mask for each center
+        g2d = grid_2d(int(2 * self.interparticle_distance), normalized=False)
+        radial_mask = g2d["r"] <= self.interparticle_distance
+        self.grid_x = g2d["x"].astype(int)[radial_mask]
+        self.grid_y = g2d["y"].astype(int)[radial_mask]
+
+        # Calculate the proper padding and create the mask
+        self.pad = int(max(self.particle_box_size, self.interparticle_distance))
+        self._mask = np.full(
+            (
+                int(self.micrograph_size + 2 * self.pad),
+                int(self.micrograph_size + 2 * self.pad),
+            ),
+            False,
+            dtype=bool,
+        )
+        self._mask_boundary = self.pad + self.boundary
+        self._mask[
+            self._mask_boundary : -1 * self._mask_boundary,
+            self._mask_boundary : -1 * self._mask_boundary,
+        ] = True
+
+        # Create the centers
+        self.centers = np.zeros(
+            (self.micrograph_count, self.particles_per_micrograph, 2), dtype=int
+        )
+        self._fail_limit = pass_threshold * self.micrograph_count
+        self._fail_count = 0
+        for i in range(self.micrograph_count):
+            self.centers[i] = self._create_centers(i)
+
+        self._clean_micrographs_accessor = _ImageAccessor(
+            self._clean_micrographs, self.micrograph_count
+        )
+        self._micrographs_accessor = _ImageAccessor(
+            self._micrographs, self.micrograph_count
+        )
+
+        self.images = _ImageAccessor(self._images, self.total_particle_count)
```

Comment:
Why not have `images` refer to the micrographs (and `clean_images` refer to the clean micrographs)? If the user wants to access the underlying particle images, they can do this using the `Simulation` object.